src/trans-word-level/show_module_hierarchy.cpp

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************\
+
+Module: Show Module Hierarchy
+
+Author: Daniel Kroening, dkr@amazon.com
+
+\*******************************************************************/
+
+#include "show_module_hierarchy.h"
+
+#include <util/namespace.h>
+
+#include <algorithm>
+
+std::size_t hierarchy_depth(const irep_idt &identifier)
+{
+  auto &as_string = id2string(identifier);
+  return std::count(as_string.begin(), as_string.end(), '.');
+}
+
+void show_module_hierarchy(
+  const symbol_table_baset &symbol_table,
+  const symbolt &main_symbol,
+  std::ostream &out)
+{
+  // find any module instances in the given module
+  std::vector<irep_idt> instances;
+
+  for(auto &symbol : symbol_table.symbols)
+    if(
+      symbol.second.type.id() == ID_module_instance &&
+      symbol.second.module == main_symbol.name)
+    {
+      instances.push_back(symbol.first);
+    }
+
+  // sort the instances alphabetically
+  std::sort(
+    instances.begin(),
+    instances.end(),
+    [](irep_idt a, irep_idt b) { return a.compare(b) < 0; });
```

Comment:
Just ideas in case you want to pick it up one of them:
1. Use a `std::set<std::string>` to represent `instances` (that's what we do in a bunch of places in CBMC)
2. Use
````c++
auto comp = [](irep_idt a, irep_idt b) { return a.compare(b) < 0; };
std::set<irep_idt, decltype(comp)> instances(comp);
````